src/simtools/production_configuration/limits_calculation.py

```diff
@@ -181,22 +225,91 @@ def compute_viewcone(self, loss_fraction):
         astropy.units.Quantity
             Viewcone radius in degrees.
         """
-        angle_to_observing_position__triggered_showers = np.column_stack(
-            [
-                self.angle_to_observing_position__triggered_showers_[name]
-                for name in self.angle_to_observing_position__triggered_showers_.dtype.names
-            ]
+        # already in radians
+        azimuth_diff = self.array_azimuth - self.shower_sim_azimuth  # * (np.pi / 180.0)
+        sim_altitude_rad = self.shower_sim_altitude  # * (np.pi / 180.0)
+        array_altitude_rad = self.array_altitude  # * (np.pi / 180.0)
+
+        x_1 = np.cos(azimuth_diff) * np.cos(sim_altitude_rad)
+        y_1 = np.sin(azimuth_diff) * np.cos(sim_altitude_rad)
+        z_1 = np.sin(sim_altitude_rad)
+        x_2 = x_1 * np.sin(array_altitude_rad) - z_1 * np.cos(array_altitude_rad)
+        y_2 = y_1
+        z_2 = x_1 * np.cos(array_altitude_rad) + z_1 * np.sin(array_altitude_rad)
+        off_angles = np.arctan2(np.sqrt(x_2**2 + y_2**2), z_2) * (180.0 / np.pi)
+
+        angle_bins = np.linspace(off_angles.min(), off_angles.max(), 200)
+        hist, _ = np.histogram(off_angles, bins=angle_bins)
+
+        _, upper_bin_edge_value = self._compute_limits(
+            hist, angle_bins, loss_fraction, axis=0, limit_type="upper"
         )
-        bin_edges, bin_edges_unit = self.get_bin_edges_and_units(
-            self.angle_to_observing_position__triggered_showers_, axis="x"
+        return upper_bin_edge_value * u.deg
+
+    def plot_data(self):
+        """Plot the core distances and energies of triggered events."""
+        num_files = len(self.list_of_files)
+        showers_per_file = len(self.simulated) // num_files
+        shower_id_triggered_adjusted = self.adjust_shower_ids_across_files(
```

Comment:
The following lines are duplicated. I suggest to write a single small service function to replace the code in both places.